gis_processor/files_db_gis.py

```diff
@@ -0,0 +1,62 @@
+from datetime import datetime
+from os import PathLike
+from pathlib import Path
+from typing import Union
+from uuid import UUID
+
+from acacore.database.files_db import FileDB
+from acacore.models.file import File
+
+
+class GisFilesDB(FileDB):
+    """Class for interacting with the `files.db` database, inherits form acacore `FileDB`."""
+
+    def __init__(self, database: Union[str, bytes, PathLike[str], PathLike[bytes]]) -> None:
+        super.__init__(database=database)
+
+    def update_rel_path(
+        self, new_rel_path: Path, uuid: Union[UUID, None] = None, old_rel_path: Union[Path, None] = None
+    ):  # noqa: E501
+        """Update the relative path of a file in the db based on either its UUID or its old relative path.
+
+        Args:
+            new_rel_path (Path): The new realtive path of the file
+            uuid (Union[UUID, None], optional): The uuid of the file. Defaults to None.
+            old_rel_path (Union[Path, None], optional): The old relative path of the file. Defaults to None.
+
+        Raises:
+            FileNotFoundError: If the file can't be found in the database based on either rel_path or uuid
+        """ """"""
+        # 1: Locate the file either using the given uuid or relative path
+        if uuid:
+            file: Union[File, None] = self.files.select(where="uuid = ?", parameters=uuid).fetchone()
```

Comment:
Parameters should be lists since they are in brackets i.e. `[str(old_rel_path)]` - or does this for some reason not work?

Reply:
Parameters are passed down to the SQLite connection, which requires an iterable object. If they are not enclosed in a list or tuple, it can cause errors (for something like int), or unforeseen behaviour (for something like str, which are iterable).